File: src/authoria/auth/static/js/validation.js
function authDataValidation(auth_data) {
    var errors = []
    
    for (const [field, value] of Object.entries(auth_data)) {
        if (value.length === 0) {
            errors.push({'field': field, 'text': 'Заполните данное поле'});
            continue
        }
        
        if (field == 'username') {
            if (username_regex.test(value) == false) {
                errors.push({'field': field, 'text': 'Некорректное имя. Используйте: a-z 0-9 _'});
            }
        }

        else if (field == 'email') {
            if (email_regex.test(value) == false) {
                errors.push({'field': field, 'text': 'Некорректный E-Mail'});
            }
        }

        else if (field == 'username_or_email') {
            if (username_regex.test(value) == false && email_regex.test(value) == false) {
                errors.push({'field': field, 'text': 'Некорректное имя или e-mail'});
            }
        }

        else if (field == 'password') {
            if (value.length < 8) {
                errors.push({'field': field, 'text': 'Сликшом короткий пароль'});
            } 
            else if (password_regex.test(value) == false) {
                errors.push({'field': field, 'text': 'Некорректный пароль'});
            }
        }

        else if (field == 'password-confirm') {
            if (!(value === auth_data['password'])) {
                errors.push({'field': 'password', 'text': 'Пароли не совпадают'});
                errors.push({'field': 'password-confirm', 'text': 'Пароли не совпадают'});
            }
        }
    }

    return errors;
}

File: src/authoria/auth/validation.py
from django.db.models import Q

from .models import User
from .encoding import checkPassword


class Validator:
    @staticmethod
    def logIn(username_or_email: str, password: str) -> list:
        errors = []

        # Check "username or email"
        try:
            user = User.objects.only('password', 'password_salt').get(
                Q(username=username_or_email) | Q(email=username_or_email)
            )

            if checkPassword(password, user.password, user.password_salt) is False:
                errors.append({
                    'field': 'password',
                    'text': 'Неверный пароль',
                })

        except User.DoesNotExist:
            errors.append({
                'field': 'username_or_email', 
                'text': 'Такого пользователя не существует'
            })

        except User.MultipleObjectsReturned:
            errors.append({
                'field': 'username_or_email', 
                'text': 'Для входа используйте имя пользователя'
            })

        return errors

    @staticmethod
    def signUp(username: str, email: str) -> list:
        errors = []

        # Check username
        if len(User.objects.filter(username=username)) > 0:
            errors.append({
                'field': 'username', 
                'text': 'Данное имя пользователя занято'
            })

        # Check email
        if len(User.objects.filter(Q(email=email) & Q(email_confirmed=True))) > 0:
            errors.append({
                'field': 'email', 
                'text': 'Данный e-mail адрес занят'
            })

        return errors